# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1236707752 14400
#      Tue Mar 10 13:55:52 2009 -0400
# Node ID d865363208d69c721c4b5e3812e215a865b21684
# Parent  e31d47f2c9bb6c6aae0e238358803f47bcf0f5ff
include <iosfwd> instead of <iostream> in header files

diff --git a/liboctave/Array.h b/liboctave/Array.h
--- a/liboctave/Array.h
+++ b/liboctave/Array.h
@@ -24,18 +24,18 @@ along with Octave; see the file COPYING.
 */
 
 #if !defined (octave_Array_h)
 #define octave_Array_h 1
 
 #include <cassert>
 #include <cstddef>
 
-#include <iostream>
 #include <algorithm>
+#include <iosfwd>
 
 #include "dim-vector.h"
 #include "idx-vector.h"
 #include "lo-traits.h"
 #include "lo-utils.h"
 #include "oct-sort.h"
 #include "quit.h"
 
diff --git a/liboctave/ArrayN.h b/liboctave/ArrayN.h
--- a/liboctave/ArrayN.h
+++ b/liboctave/ArrayN.h
@@ -20,22 +20,22 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ArrayN_h)
 #define octave_ArrayN_h 1
 
-#include <iostream>
-
 #include <cassert>
 #include <climits>
 #include <cstdlib>
 
+#include <iosfwd>
+
 #include "Array.h"
 #include "Array2.h"
 #include "lo-error.h"
 #include "lo-math.h"
 
 class idx_vector;
 
 // N-dimensional array class.
diff --git a/liboctave/Bounds.h b/liboctave/Bounds.h
--- a/liboctave/Bounds.h
+++ b/liboctave/Bounds.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_Bounds_h)
 #define octave_Bounds_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dColVector.h"
 
 class
 Bounds
 {
 public:
 
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,22 @@
+2009-03-10  John W. Eaton  <jwe@octave.org>
+
+	* Array.h, ArrayN.h, Bounds.h, CmplxAEPBAL.h, CmplxCHOL.h,
+	CmplxGEPBAL.h, CmplxHESS.h, CmplxQR.h, CmplxQRP.h, CmplxSCHUR.h,
+	CmplxSVD.h, CollocWt.h, EIG.h, FEGrid.h, LinConst.h, Range.h,
+	Sparse.h, SparseCmplxQR.h, SparseQR.h, dbleAEPBAL.h, dbleCHOL.h,
+	dbleGEPBAL.h, dbleHESS.h, dbleQR.h, dbleQRP.h, dbleSCHUR.h,
+	dbleSVD.h, dim-vector.h, fCmplxAEPBAL.h, fCmplxCHOL.h,
+	fCmplxGEPBAL.h, fCmplxHESS.h, fCmplxQR.h, fCmplxQRP.h,
+	fCmplxSCHUR.h, fCmplxSVD.h, fEIG.h, floatAEPBAL.h, floatCHOL.h,
+	floatGEPBAL.h, floatHESS.h, floatQR.h, floatQRP.h, floatSCHUR.h,
+	floatSVD.h, idx-vector.h, lo-utils.h, oct-inttypes.h,
+	oct-spparms.h, str-vec.h: Include <iosfwd> instead of <iosstream>.
+
 2009-03-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* mx-inlines.cc (OP_CUMMINMAX_FCN2): r -> r0 where appropriate.
 
 2009-03-08  Jaroslav Hajek  <highegg@gmail.com>
 
 	* idx-vector.h (idx_vector::bloop): loop --> bloop.
 	(idx_vector::loop): New method.
diff --git a/liboctave/CmplxAEPBAL.h b/liboctave/CmplxAEPBAL.h
--- a/liboctave/CmplxAEPBAL.h
+++ b/liboctave/CmplxAEPBAL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexAEPBALANCE_h)
 #define octave_ComplexAEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "base-aepbal.h"
 #include "CMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/CmplxCHOL.h b/liboctave/CmplxCHOL.h
--- a/liboctave/CmplxCHOL.h
+++ b/liboctave/CmplxCHOL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexCHOL_h)
 #define octave_ComplexCHOL_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "CMatrix.h"
 #include "CColVector.h"
 
 class
 OCTAVE_API
 ComplexCHOL
 {
diff --git a/liboctave/CmplxGEPBAL.h b/liboctave/CmplxGEPBAL.h
--- a/liboctave/CmplxGEPBAL.h
+++ b/liboctave/CmplxGEPBAL.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexGEPBALANCE_h)
 #define octave_ComplexGEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "CMatrix.h"
 #include "dMatrix.h"
 
 class
 OCTAVE_API
 ComplexGEPBALANCE
diff --git a/liboctave/CmplxHESS.h b/liboctave/CmplxHESS.h
--- a/liboctave/CmplxHESS.h
+++ b/liboctave/CmplxHESS.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexHESS_h)
 #define octave_ComplexHESS_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "CMatrix.h"
 
 class
 OCTAVE_API
 ComplexHESS
 {
 public:
diff --git a/liboctave/CmplxQR.h b/liboctave/CmplxQR.h
--- a/liboctave/CmplxQR.h
+++ b/liboctave/CmplxQR.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexQR_h)
 #define octave_ComplexQR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "CMatrix.h"
 #include "CColVector.h"
 #include "CRowVector.h"
 #include "dbleQR.h"
 
 
 class
diff --git a/liboctave/CmplxQRP.h b/liboctave/CmplxQRP.h
--- a/liboctave/CmplxQRP.h
+++ b/liboctave/CmplxQRP.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexQRP_h)
 #define octave_ComplexQRP_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "CmplxQR.h"
 #include "PermMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
 ComplexQRP : public ComplexQR
diff --git a/liboctave/CmplxSCHUR.h b/liboctave/CmplxSCHUR.h
--- a/liboctave/CmplxSCHUR.h
+++ b/liboctave/CmplxSCHUR.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexSCHUR_h)
 #define octave_ComplexSCHUR_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "CMatrix.h"
 
 class
 OCTAVE_API
 ComplexSCHUR
 {
diff --git a/liboctave/CmplxSVD.h b/liboctave/CmplxSVD.h
--- a/liboctave/CmplxSVD.h
+++ b/liboctave/CmplxSVD.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ComplexSVD_h)
 #define octave_ComplexSVD_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dDiagMatrix.h"
 #include "CMatrix.h"
 #include "dbleSVD.h"
 
 class
 OCTAVE_API
 ComplexSVD
diff --git a/liboctave/CollocWt.h b/liboctave/CollocWt.h
--- a/liboctave/CollocWt.h
+++ b/liboctave/CollocWt.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_CollocWt_h)
 #define octave_CollocWt_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
 CollocWt
 {
diff --git a/liboctave/EIG.h b/liboctave/EIG.h
--- a/liboctave/EIG.h
+++ b/liboctave/EIG.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_EIG_h)
 #define octave_EIG_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "CMatrix.h"
 #include "CColVector.h"
 
 class
 OCTAVE_API
 EIG
diff --git a/liboctave/FEGrid.h b/liboctave/FEGrid.h
--- a/liboctave/FEGrid.h
+++ b/liboctave/FEGrid.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FEGrid_h)
 #define octave_FEGrid_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dColVector.h"
 
 class
 FEGrid
 {
 public:
 
diff --git a/liboctave/LinConst.h b/liboctave/LinConst.h
--- a/liboctave/LinConst.h
+++ b/liboctave/LinConst.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_LinConst_h)
 #define octave_LinConst_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 class ColumnVector;
 
 #include <cfloat>
 
 #include "dMatrix.h"
 #include "Bounds.h"
 
diff --git a/liboctave/Range.h b/liboctave/Range.h
--- a/liboctave/Range.h
+++ b/liboctave/Range.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_Range_h)
 #define octave_Range_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "oct-sort.h"
 
 class
 OCTAVE_API
 Range
 {
diff --git a/liboctave/Sparse.h b/liboctave/Sparse.h
--- a/liboctave/Sparse.h
+++ b/liboctave/Sparse.h
@@ -23,17 +23,17 @@ along with Octave; see the file COPYING.
 */
 
 #if !defined (octave_Sparse_h)
 #define octave_Sparse_h 1
 
 #include <cassert>
 #include <cstddef>
 
-#include <iostream>
+#include <iosfwd>
 
 #include "Array.h"
 #include "Array2.h"
 #include "dim-vector.h"
 #include "lo-utils.h"
 
 #include "oct-sort.h"
 
diff --git a/liboctave/SparseCmplxQR.h b/liboctave/SparseCmplxQR.h
--- a/liboctave/SparseCmplxQR.h
+++ b/liboctave/SparseCmplxQR.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (sparse_cmplx_QR_h)
 #define sparse_cmplx_QR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "CMatrix.h"
 #include "dSparse.h"
 #include "CSparse.h"
 #include "oct-sparse.h"
 
 #ifdef IDX_TYPE_LONG
diff --git a/liboctave/SparseQR.h b/liboctave/SparseQR.h
--- a/liboctave/SparseQR.h
+++ b/liboctave/SparseQR.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (sparse_QR_h)
 #define sparse_QR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "CMatrix.h"
 #include "dSparse.h"
 #include "CSparse.h"
 #include "oct-sparse.h"
 
 #ifdef IDX_TYPE_LONG
diff --git a/liboctave/dbleAEPBAL.h b/liboctave/dbleAEPBAL.h
--- a/liboctave/dbleAEPBAL.h
+++ b/liboctave/dbleAEPBAL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_AEPBALANCE_h)
 #define octave_AEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "base-aepbal.h"
 #include "dMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/dbleCHOL.h b/liboctave/dbleCHOL.h
--- a/liboctave/dbleCHOL.h
+++ b/liboctave/dbleCHOL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_CHOL_h)
 #define octave_CHOL_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
 CHOL
 {
diff --git a/liboctave/dbleGEPBAL.h b/liboctave/dbleGEPBAL.h
--- a/liboctave/dbleGEPBAL.h
+++ b/liboctave/dbleGEPBAL.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_GEPBALANCE_h)
 #define octave_GEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "dMatrix.h"
 
 class
 OCTAVE_API
 GEPBALANCE
 {
diff --git a/liboctave/dbleHESS.h b/liboctave/dbleHESS.h
--- a/liboctave/dbleHESS.h
+++ b/liboctave/dbleHESS.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_HESS_h)
 #define octave_HESS_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 
 class
 OCTAVE_API
 HESS
 {
 public:
diff --git a/liboctave/dbleQR.h b/liboctave/dbleQR.h
--- a/liboctave/dbleQR.h
+++ b/liboctave/dbleQR.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_QR_h)
 #define octave_QR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dMatrix.h"
 #include "dColVector.h"
 #include "dRowVector.h"
 
 class
 OCTAVE_API
 QR
diff --git a/liboctave/dbleQRP.h b/liboctave/dbleQRP.h
--- a/liboctave/dbleQRP.h
+++ b/liboctave/dbleQRP.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_QRP_h)
 #define octave_QRP_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dbleQR.h"
 #include "PermMatrix.h"
 #include "dColVector.h"
 
 class
 OCTAVE_API
 QRP : public QR
diff --git a/liboctave/dbleSCHUR.h b/liboctave/dbleSCHUR.h
--- a/liboctave/dbleSCHUR.h
+++ b/liboctave/dbleSCHUR.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_SCHUR_h)
 #define octave_SCHUR_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "dMatrix.h"
 
 class
 OCTAVE_API
 SCHUR
 {
diff --git a/liboctave/dbleSVD.h b/liboctave/dbleSVD.h
--- a/liboctave/dbleSVD.h
+++ b/liboctave/dbleSVD.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_SVD_h)
 #define octave_SVD_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "dDiagMatrix.h"
 #include "dMatrix.h"
 
 class
 OCTAVE_API
 SVD
 {
diff --git a/liboctave/dim-vector.h b/liboctave/dim-vector.h
--- a/liboctave/dim-vector.h
+++ b/liboctave/dim-vector.h
@@ -19,16 +19,17 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_dim_vector_h)
 #define octave_dim_vector_h 1
 
 #include <cassert>
+
 #include <sstream>
 #include <string>
 
 #include "lo-error.h"
 #include "oct-types.h"
 
 class
 dim_vector
diff --git a/liboctave/fCmplxAEPBAL.h b/liboctave/fCmplxAEPBAL.h
--- a/liboctave/fCmplxAEPBAL.h
+++ b/liboctave/fCmplxAEPBAL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexAEPBALANCE_h)
 #define octave_FloatComplexAEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "base-aepbal.h"
 #include "fCMatrix.h"
 #include "fColVector.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/fCmplxCHOL.h b/liboctave/fCmplxCHOL.h
--- a/liboctave/fCmplxCHOL.h
+++ b/liboctave/fCmplxCHOL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexCHOL_h)
 #define octave_FloatComplexCHOL_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fCMatrix.h"
 #include "fCColVector.h"
 
 class
 OCTAVE_API
 FloatComplexCHOL
 {
diff --git a/liboctave/fCmplxGEPBAL.h b/liboctave/fCmplxGEPBAL.h
--- a/liboctave/fCmplxGEPBAL.h
+++ b/liboctave/fCmplxGEPBAL.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexGEPBALANCE_h)
 #define octave_FloatComplexGEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "fMatrix.h"
 #include "fCMatrix.h"
 
 class
 OCTAVE_API
 FloatComplexGEPBALANCE
diff --git a/liboctave/fCmplxHESS.h b/liboctave/fCmplxHESS.h
--- a/liboctave/fCmplxHESS.h
+++ b/liboctave/fCmplxHESS.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexHESS_h)
 #define octave_FloatComplexHESS_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fCMatrix.h"
 
 class
 OCTAVE_API
 FloatComplexHESS
 {
 public:
diff --git a/liboctave/fCmplxQR.h b/liboctave/fCmplxQR.h
--- a/liboctave/fCmplxQR.h
+++ b/liboctave/fCmplxQR.h
@@ -22,17 +22,17 @@ along with Octave; see the file COPYING.
 
 */
 
 // updating/downdating by Jaroslav Hajek 2008
 
 #if !defined (octave_FloatComplexQR_h)
 #define octave_FloatComplexQR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fCMatrix.h"
 #include "fCColVector.h"
 #include "fCRowVector.h"
 #include "dbleQR.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/fCmplxQRP.h b/liboctave/fCmplxQRP.h
--- a/liboctave/fCmplxQRP.h
+++ b/liboctave/fCmplxQRP.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexQRP_h)
 #define octave_FloatComplexQRP_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fCmplxQR.h"
 #include "PermMatrix.h"
 #include "fColVector.h"
 
 class
 OCTAVE_API
 FloatComplexQRP : public FloatComplexQR
diff --git a/liboctave/fCmplxSCHUR.h b/liboctave/fCmplxSCHUR.h
--- a/liboctave/fCmplxSCHUR.h
+++ b/liboctave/fCmplxSCHUR.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexSCHUR_h)
 #define octave_FloatComplexSCHUR_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "fCMatrix.h"
 
 class
 OCTAVE_API
 FloatComplexSCHUR
 {
diff --git a/liboctave/fCmplxSVD.h b/liboctave/fCmplxSVD.h
--- a/liboctave/fCmplxSVD.h
+++ b/liboctave/fCmplxSVD.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatComplexSVD_h)
 #define octave_FloatComplexSVD_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fDiagMatrix.h"
 #include "fCMatrix.h"
 #include "dbleSVD.h"
 
 class
 OCTAVE_API
 FloatComplexSVD
diff --git a/liboctave/fEIG.h b/liboctave/fEIG.h
--- a/liboctave/fEIG.h
+++ b/liboctave/fEIG.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_float_EIG_h)
 #define octave_float_EIG_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fMatrix.h"
 #include "fCMatrix.h"
 #include "fCColVector.h"
 
 class
 OCTAVE_API
 FloatEIG
diff --git a/liboctave/floatAEPBAL.h b/liboctave/floatAEPBAL.h
--- a/liboctave/floatAEPBAL.h
+++ b/liboctave/floatAEPBAL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatAEPBALANCE_h)
 #define octave_FloatAEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "base-aepbal.h"
 #include "fMatrix.h"
 #include "fColVector.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/floatCHOL.h b/liboctave/floatCHOL.h
--- a/liboctave/floatCHOL.h
+++ b/liboctave/floatCHOL.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatCHOL_h)
 #define octave_FloatCHOL_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fMatrix.h"
 #include "fColVector.h"
 
 class
 OCTAVE_API
 FloatCHOL
 {
diff --git a/liboctave/floatGEPBAL.h b/liboctave/floatGEPBAL.h
--- a/liboctave/floatGEPBAL.h
+++ b/liboctave/floatGEPBAL.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatGEPBALANCE_h)
 #define octave_FloatGEPBALANCE_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "fMatrix.h"
 
 class
 OCTAVE_API
 FloatGEPBALANCE
 {
diff --git a/liboctave/floatHESS.h b/liboctave/floatHESS.h
--- a/liboctave/floatHESS.h
+++ b/liboctave/floatHESS.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatHESS_h)
 #define octave_FloatHESS_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fMatrix.h"
 
 class
 OCTAVE_API
 FloatHESS
 {
 public:
diff --git a/liboctave/floatQR.h b/liboctave/floatQR.h
--- a/liboctave/floatQR.h
+++ b/liboctave/floatQR.h
@@ -20,17 +20,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatQR_h)
 #define octave_FloatQR_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fMatrix.h"
 #include "fColVector.h"
 #include "fRowVector.h"
 #include "dbleQR.h"
 
 class
 OCTAVE_API
diff --git a/liboctave/floatQRP.h b/liboctave/floatQRP.h
--- a/liboctave/floatQRP.h
+++ b/liboctave/floatQRP.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatQRP_h)
 #define octave_FloatQRP_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "floatQR.h"
 #include "PermMatrix.h"
 #include "fColVector.h"
 
 class
 OCTAVE_API
 FloatQRP : public FloatQR
diff --git a/liboctave/floatSCHUR.h b/liboctave/floatSCHUR.h
--- a/liboctave/floatSCHUR.h
+++ b/liboctave/floatSCHUR.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatSCHUR_h)
 #define octave_FloatSCHUR_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "fMatrix.h"
 
 class
 OCTAVE_API
 FloatSCHUR
 {
diff --git a/liboctave/floatSVD.h b/liboctave/floatSVD.h
--- a/liboctave/floatSVD.h
+++ b/liboctave/floatSVD.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_FloatSVD_h)
 #define octave_FloatSVD_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "fDiagMatrix.h"
 #include "fMatrix.h"
 #include "dbleSVD.h"
 
 class
 OCTAVE_API
 FloatSVD
diff --git a/liboctave/idx-vector.h b/liboctave/idx-vector.h
--- a/liboctave/idx-vector.h
+++ b/liboctave/idx-vector.h
@@ -20,19 +20,20 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_idx_vector_h)
 #define octave_idx_vector_h 1
 
-#include <iostream>
+#include <cassert>
+
 #include <algorithm>
-#include <cassert>
+#include <iosfwd>
 
 #include "dim-vector.h"
 #include "oct-inttypes.h"
 #include "oct-alloc.h"
 
 template<class T> class Array;
 template<class T> class Sparse;
 class Range;
diff --git a/liboctave/lo-utils.h b/liboctave/lo-utils.h
--- a/liboctave/lo-utils.h
+++ b/liboctave/lo-utils.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_liboctave_utils_h)
 #define octave_liboctave_utils_h 1
 
 #include <cstdio>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "oct-cmplx.h"
 #include "oct-types.h"
 #include "syswait.h"
 
 extern OCTAVE_API octave_idx_type NINTbig (double x);
 extern OCTAVE_API octave_idx_type NINTbig (float x);
diff --git a/liboctave/oct-inttypes.h b/liboctave/oct-inttypes.h
--- a/liboctave/oct-inttypes.h
+++ b/liboctave/oct-inttypes.h
@@ -23,17 +23,17 @@ along with Octave; see the file COPYING.
 
 #if !defined (octave_inttypes_h)
 #define octave_inttypes_h 1
 
 #include <climits>
 #include <cstdlib>
 
 #include <limits>
-#include <iostream>
+#include <iosfwd>
 
 #include "lo-traits.h"
 #include "lo-math.h"
 #include "oct-types.h"
 #include "lo-ieee.h"
 #include "lo-mappers.h"
 
 #ifdef OCTAVE_INT_USE_LONG_DOUBLE
diff --git a/liboctave/oct-spparms.h b/liboctave/oct-spparms.h
--- a/liboctave/oct-spparms.h
+++ b/liboctave/oct-spparms.h
@@ -22,17 +22,17 @@ along with Octave; see the file COPYING.
 */
 
 #if !defined (octave_oct_spparms_h)
 #define octave_oct_spparms_h 1
 
 #include <cassert>
 #include <cstddef>
 
-#include <iostream>
+#include <iosfwd>
 
 #include "str-vec.h"
 #include "dColVector.h"
 #include "dNDArray.h"
 
 #define OCTAVE_SPARSE_CONTROLS_SIZE 13
 
 class
diff --git a/liboctave/str-vec.h b/liboctave/str-vec.h
--- a/liboctave/str-vec.h
+++ b/liboctave/str-vec.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_str_vec_h)
 #define octave_str_vec_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <list>
 #include <string>
 
 #include "Array.h"
 
 class
 OCTAVE_API
 string_vector : public Array<std::string>
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,26 @@
+2009-03-10  John W. Eaton  <jwe@octave.org>
+
+	* c-file-ptr-stream.cc, dynamic-ld.cc, error.cc, lex.l, pager.cc,
+	zfstream.cc: Include <iostream>.
+	* zfstream.h: Include <iosfwd> instead of <istream> and <ostream>.
+	* c-file-ptr-stream.h, help.h, load-path.h, load-save.h,
+	ls-ascii-helper.h, oct-iostrm.h, oct-stream.h, ov-base-diag.h,
+	ov-base-int.h, ov-base-mat.h, ov-base-scalar.h, ov-base-sparse.h,
+	ov-base.h, ov-bool-mat.h, ov-bool-sparse.h, ov-bool.h, ov-cell.h,
+	ov-ch-mat.h, ov-class.h, ov-colon.h, ov-complex.h, ov-cs-list.h,
+	ov-cx-mat.h, ov-cx-sparse.h, ov-fcn-handle.h, ov-fcn-inline.h,
+	ov-float.h, ov-flt-complex.h, ov-flt-cx-mat.h, ov-flt-re-mat.h,
+	ov-intx.h, ov-list.h, ov-range.h, ov-re-mat.h, ov-re-sparse.h,
+	ov-scalar.h, ov-str-mat.h, ov-struct.h, ov.h, pager.h,
+	pr-output.h, procstream.h, pt-assign.h, pt-cell.h, pt-const.h,
+	pt-fcn-handle.h, pt-id.h, pt-mat.h, pt.h, utils.h:
+	Include <iosfwd> instead of <iosstream>.
+
 2009-03-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Fsize_equal): Allow single argument call.
 	* DLD-FUNCTIONS/max.cc (Fcummin, Fcummax): Update docs.
 
 2009-03-09  Benjamin Lindner  <lindnerb@users.sourceforge.net>
 
 	* ls-ascii-helper.h, ls-ascii-helper.cc: New files.
diff --git a/src/c-file-ptr-stream.cc b/src/c-file-ptr-stream.cc
--- a/src/c-file-ptr-stream.cc
+++ b/src/c-file-ptr-stream.cc
@@ -20,16 +20,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+#include <iostream>
+
 #include "c-file-ptr-stream.h"
 
 #ifndef SEEK_SET
 #define SEEK_SET 0
 #endif
 
 #ifndef SEEK_CUR
 #define SEEK_CUR 1
diff --git a/src/c-file-ptr-stream.h b/src/c-file-ptr-stream.h
--- a/src/c-file-ptr-stream.h
+++ b/src/c-file-ptr-stream.h
@@ -19,19 +19,20 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_c_file_ptr_stream_h)
 #define octave_c_file_ptr_stream_h 1
 
-#include <iostream>
 #include <cstdio>
 
+#include <iosfwd>
+
 class
 c_file_ptr_buf : public std::streambuf
 {
 public:
 
 #if !defined (CXX_ISO_COMPLIANT_LIBRARY)
   typedef int int_type;
 #else
diff --git a/src/dynamic-ld.cc b/src/dynamic-ld.cc
--- a/src/dynamic-ld.cc
+++ b/src/dynamic-ld.cc
@@ -20,16 +20,17 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+#include <iostream>
 #include <list>
 
 #include "oct-env.h"
 #include "oct-time.h"
 #include "file-stat.h"
 
 #include <defaults.h>
 
diff --git a/src/error.cc b/src/error.cc
--- a/src/error.cc
+++ b/src/error.cc
@@ -23,16 +23,17 @@ along with Octave; see the file COPYING.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cstdarg>
 #include <cstring>
 
+#include <iostream>
 #include <sstream>
 #include <string>
 
 #include "defun.h"
 #include "error.h"
 #include "input.h"
 #include "pager.h"
 #include "oct-obj.h"
diff --git a/src/help.h b/src/help.h
--- a/src/help.h
+++ b/src/help.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_help_h)
 #define octave_help_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 class string_vector;
 
 extern string_vector make_name_list (void);
 
 extern OCTINTERP_API std::string raw_help (const std::string&, bool&);
 
diff --git a/src/lex.l b/src/lex.l
--- a/src/lex.l
+++ b/src/lex.l
@@ -35,16 +35,17 @@ along with Octave; see the file COPYING.
 %{
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cctype>
 #include <cstring>
 
+#include <iostream>
 #include <set>
 #include <sstream>
 #include <string>
 #include <stack>
 
 #ifdef HAVE_UNISTD_H
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
diff --git a/src/load-path.h b/src/load-path.h
--- a/src/load-path.h
+++ b/src/load-path.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_load_path_h)
 #define octave_load_path_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <list>
 #include <map>
 #include <string>
 
 #include "pathsearch.h"
 #include "str-vec.h"
 
 class
diff --git a/src/load-save.h b/src/load-save.h
--- a/src/load-save.h
+++ b/src/load-save.h
@@ -19,18 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_load_save_h)
 #define octave_load_save_h 1
 
-#include <iostream>
-
+#include <iosfwd>
 #include <string>
 
 class octave_value;
 
 // FIXME: maybe MAT5 and MAT7 should be options to MAT_BINARY.
 // Similarly, save_as_floats may be an option for LS_BINARY, LS_HDF5 etc.
 enum load_save_format_type
   {
diff --git a/src/ls-ascii-helper.h b/src/ls-ascii-helper.h
--- a/src/ls-ascii-helper.h
+++ b/src/ls-ascii-helper.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_ls_ascii_helper_h)
 #define octave_ls_ascii_helper_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 extern OCTINTERP_API void
 skip_until_newline (std::istream& is, bool keep_newline = false);
 
 extern OCTINTERP_API void
 skip_preceeding_newline (std::istream& is);
 
diff --git a/src/oct-iostrm.h b/src/oct-iostrm.h
--- a/src/oct-iostrm.h
+++ b/src/oct-iostrm.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_octave_iostream_h)
 #define octave_octave_iostream_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "oct-stream.h"
 
 class
 octave_base_iostream : public octave_base_stream
 {
 public:
 
diff --git a/src/oct-stream.h b/src/oct-stream.h
--- a/src/oct-stream.h
+++ b/src/oct-stream.h
@@ -24,17 +24,17 @@ along with Octave; see the file COPYING.
 #if !defined (octave_octave_stream_h)
 #define octave_octave_stream_h 1
 
 class Matrix;
 class string_vector;
 class octave_value;
 class octave_value_list;
 
-#include <iostream>
+#include <iosfwd>
 #include <sstream>
 #include <string>
 #include <map>
 
 #include "Array.h"
 #include "data-conv.h"
 #include "lo-utils.h"
 #include "mach-info.h"
diff --git a/src/ov-base-diag.h b/src/ov-base-diag.h
--- a/src/ov-base-diag.h
+++ b/src/ov-base-diag.h
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_diag_h)
 #define octave_base_diag_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "oct-obj.h"
 #include "ov-base.h"
 #include "ov-typeinfo.h"
diff --git a/src/ov-base-int.h b/src/ov-base-int.h
--- a/src/ov-base-int.h
+++ b/src/ov-base-int.h
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_int_matrix_h)
 #define octave_base_int_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "ov-base.h"
diff --git a/src/ov-base-mat.h b/src/ov-base-mat.h
--- a/src/ov-base-mat.h
+++ b/src/ov-base-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_matrix_h)
 #define octave_base_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 #include "MatrixType.h"
 
 #include "error.h"
 #include "oct-obj.h"
diff --git a/src/ov-base-scalar.h b/src/ov-base-scalar.h
--- a/src/ov-base-scalar.h
+++ b/src/ov-base-scalar.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_scalar_h)
 #define octave_base_scalar_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-mappers.h"
 #include "lo-utils.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 #include "MatrixType.h"
 
diff --git a/src/ov-base-sparse.h b/src/ov-base-sparse.h
--- a/src/ov-base-sparse.h
+++ b/src/ov-base-sparse.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_sparse_h)
 #define octave_base_sparse_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-obj.h"
 #include "ov-base.h"
 #include "ov-typeinfo.h"
diff --git a/src/ov-base.h b/src/ov-base.h
--- a/src/ov-base.h
+++ b/src/ov-base.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_base_value_h)
 #define octave_base_value_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <list>
 #include <string>
 
 #include "Range.h"
 #include "data-conv.h"
 #include "mxarray.h"
 #include "mx-base.h"
 #include "str-vec.h"
diff --git a/src/ov-bool-mat.h b/src/ov-bool-mat.h
--- a/src/ov-bool-mat.h
+++ b/src/ov-bool-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_bool_matrix_h)
 #define octave_bool_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 
 #include "error.h"
 #include "oct-stream.h"
 #include "ov-base.h"
diff --git a/src/ov-bool-sparse.h b/src/ov-bool-sparse.h
--- a/src/ov-bool-sparse.h
+++ b/src/ov-bool-sparse.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_sparse_bool_matrix_h)
 #define octave_sparse_bool_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-bool.h b/src/ov-bool.h
--- a/src/ov-bool.h
+++ b/src/ov-bool.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_bool_h)
 #define octave_bool_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-utils.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "oct-stream.h"
diff --git a/src/ov-cell.h b/src/ov-cell.h
--- a/src/ov-cell.h
+++ b/src/ov-cell.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_cell_h)
 #define octave_cell_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "Cell.h"
 #include "error.h"
diff --git a/src/ov-ch-mat.h b/src/ov-ch-mat.h
--- a/src/ov-ch-mat.h
+++ b/src/ov-ch-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_char_matrix_h)
 #define octave_char_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "ov-base.h"
diff --git a/src/ov-class.h b/src/ov-class.h
--- a/src/ov-class.h
+++ b/src/ov-class.h
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_class_h)
 #define octave_class_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-alloc.h"
 #include "oct-map.h"
diff --git a/src/ov-colon.h b/src/ov-colon.h
--- a/src/ov-colon.h
+++ b/src/ov-colon.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_magic_colon_h)
 #define octave_magic_colon_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "ov-base.h"
 #include "ov-typeinfo.h"
diff --git a/src/ov-complex.h b/src/ov-complex.h
--- a/src/ov-complex.h
+++ b/src/ov-complex.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_complex_h)
 #define octave_complex_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-ieee.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
diff --git a/src/ov-cs-list.h b/src/ov-cs-list.h
--- a/src/ov-cs-list.h
+++ b/src/ov-cs-list.h
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_cs_list_h)
 #define octave_cs_list_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "Cell.h"
 #include "error.h"
 #include "oct-alloc.h"
diff --git a/src/ov-cx-mat.h b/src/ov-cx-mat.h
--- a/src/ov-cx-mat.h
+++ b/src/ov-cx-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_complex_matrix_h)
 #define octave_complex_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-cx-sparse.h b/src/ov-cx-sparse.h
--- a/src/ov-cx-sparse.h
+++ b/src/ov-cx-sparse.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_sparse_complex_matrix_h)
 #define octave_sparse_complex_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-fcn-handle.h b/src/ov-fcn-handle.h
--- a/src/ov-fcn-handle.h
+++ b/src/ov-fcn-handle.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_fcn_handle_h)
 #define octave_fcn_handle_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "oct-alloc.h"
 
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-fcn.h"
 #include "ov-typeinfo.h"
diff --git a/src/ov-fcn-inline.h b/src/ov-fcn-inline.h
--- a/src/ov-fcn-inline.h
+++ b/src/ov-fcn-inline.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_fcn_inline_h)
 #define octave_fcn_inline_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "oct-alloc.h"
 
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-fcn.h"
 #include "ov-typeinfo.h"
diff --git a/src/ov-float.h b/src/ov-float.h
--- a/src/ov-float.h
+++ b/src/ov-float.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_float_h)
 #define octave_float_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-ieee.h"
 #include "lo-mappers.h"
 #include "lo-utils.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
diff --git a/src/ov-flt-complex.h b/src/ov-flt-complex.h
--- a/src/ov-flt-complex.h
+++ b/src/ov-flt-complex.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_float_complex_h)
 #define octave_float_complex_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-ieee.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
diff --git a/src/ov-flt-cx-mat.h b/src/ov-flt-cx-mat.h
--- a/src/ov-flt-cx-mat.h
+++ b/src/ov-flt-cx-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_float_complex_matrix_h)
 #define octave_float_complex_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-flt-re-mat.h b/src/ov-flt-re-mat.h
--- a/src/ov-flt-re-mat.h
+++ b/src/ov-flt-re-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_float_matrix_h)
 #define octave_float_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-intx.h b/src/ov-intx.h
--- a/src/ov-intx.h
+++ b/src/ov-intx.h
@@ -17,17 +17,17 @@ for more details.
 You should have received a copy of the GNU General Public License
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-list.h b/src/ov-list.h
--- a/src/ov-list.h
+++ b/src/ov-list.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_list_h)
 #define octave_list_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "Cell.h"
 #include "error.h"
 #include "oct-alloc.h"
diff --git a/src/ov-range.h b/src/ov-range.h
--- a/src/ov-range.h
+++ b/src/ov-range.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_range_h)
 #define octave_range_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "Range.h"
 
 #include "lo-mappers.h"
 #include "lo-utils.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
diff --git a/src/ov-re-mat.h b/src/ov-re-mat.h
--- a/src/ov-re-mat.h
+++ b/src/ov-re-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_matrix_h)
 #define octave_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-re-sparse.h b/src/ov-re-sparse.h
--- a/src/ov-re-sparse.h
+++ b/src/ov-re-sparse.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_sparse_matrix_h)
 #define octave_sparse_matrix_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
diff --git a/src/ov-scalar.h b/src/ov-scalar.h
--- a/src/ov-scalar.h
+++ b/src/ov-scalar.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_scalar_h)
 #define octave_scalar_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "lo-ieee.h"
 #include "lo-mappers.h"
 #include "lo-utils.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
diff --git a/src/ov-str-mat.h b/src/ov-str-mat.h
--- a/src/ov-str-mat.h
+++ b/src/ov-str-mat.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_char_matrix_str_h)
 #define octave_char_matrix_str_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-stream.h"
 #include "ov.h"
diff --git a/src/ov-struct.h b/src/ov-struct.h
--- a/src/ov-struct.h
+++ b/src/ov-struct.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_struct_h)
 #define octave_struct_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-alloc.h"
 #include "oct-map.h"
diff --git a/src/ov.h b/src/ov.h
--- a/src/ov.h
+++ b/src/ov.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_value_h)
 #define octave_value_h 1
 
 #include <cstdlib>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 #include <list>
 
 #include "Range.h"
 #include "data-conv.h"
 #include "idx-vector.h"
 #include "mach-info.h"
 #include "mxarray.h"
diff --git a/src/pager.cc b/src/pager.cc
--- a/src/pager.cc
+++ b/src/pager.cc
@@ -21,16 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <fstream>
+#include <iostream>
 #include <string>
 
 #include "cmd-edit.h"
 #include "oct-env.h"
 
 #include "procstream.h"
 
 #include <defaults.h>
diff --git a/src/pager.h b/src/pager.h
--- a/src/pager.h
+++ b/src/pager.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_pager_h)
 #define octave_pager_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <sstream>
 #include <string>
 
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
 
 class
diff --git a/src/parse.y b/src/parse.y
--- a/src/parse.y
+++ b/src/parse.y
@@ -34,16 +34,17 @@ along with Octave; see the file COPYING.
 
 #include <cassert>
 #include <cstdio>
 
 #ifdef YYBYACC
 #include <cstdlib>
 #endif
 
+#include <iostream>
 #include <map>
 #include <sstream>
 
 #include "Cell.h"
 #include "Matrix.h"
 #include "cmd-edit.h"
 #include "cmd-hist.h"
 #include "file-ops.h"
diff --git a/src/pr-output.h b/src/pr-output.h
--- a/src/pr-output.h
+++ b/src/pr-output.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_pr_output_h)
 #define octave_pr_output_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 #include "oct-cmplx.h"
 
 template <typename T> class ArrayN;
 class ComplexMatrix;
 class FloatComplexMatrix;
 class ComplexDiagMatrix;
 class FloatComplexDiagMatrix;
diff --git a/src/procstream.cc b/src/procstream.cc
--- a/src/procstream.cc
+++ b/src/procstream.cc
@@ -20,16 +20,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+#include <iostream>
+
 #include "procstream.h"
 
 procstreambase::procstreambase (const std::string& command, int mode)
 {
   pb_init ();
 
   if (! pb.open (command.c_str (), mode))
     std::ios::setstate (std::ios::badbit);
diff --git a/src/procstream.h b/src/procstream.h
--- a/src/procstream.h
+++ b/src/procstream.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_procstream_h)
 #define octave_procstream_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
 
 #include "oct-procbuf.h"
 
diff --git a/src/pt-assign.h b/src/pt-assign.h
--- a/src/pt-assign.h
+++ b/src/pt-assign.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_assign_h)
 #define octave_tree_assign_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 class tree_argument_list;
 class tree_walker;
 
 class octave_value;
 class octave_value_list;
 class octave_lvalue;
diff --git a/src/pt-cell.h b/src/pt-cell.h
--- a/src/pt-cell.h
+++ b/src/pt-cell.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_cell_h)
 #define octave_tree_cell_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 class octave_value;
 class octave_value_list;
 class tree_argument_list;
 
 class tree_walker;
 
 #include "pt-mat.h"
diff --git a/src/pt-const.h b/src/pt-const.h
--- a/src/pt-const.h
+++ b/src/pt-const.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_const_h)
 #define octave_tree_const_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "oct-alloc.h"
 
 class octave_value_list;
 class tree_walker;
 
 #include "ov.h"
diff --git a/src/pt-fcn-handle.h b/src/pt-fcn-handle.h
--- a/src/pt-fcn-handle.h
+++ b/src/pt-fcn-handle.h
@@ -18,17 +18,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_fcn_handle_h)
 #define octave_fcn_handle_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "pt-bp.h"
 #include "pt-exp.h"
 #include "pt-misc.h"
 #include "pt-stmt.h"
 #include "symtab.h"
 
diff --git a/src/pt-id.h b/src/pt-id.h
--- a/src/pt-id.h
+++ b/src/pt-id.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_identifier_h)
 #define octave_tree_identifier_h 1
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 class octave_value;
 class octave_value_list;
 class octave_function;
 
 class tree_walker;
 
diff --git a/src/pt-mat.h b/src/pt-mat.h
--- a/src/pt-mat.h
+++ b/src/pt-mat.h
@@ -19,17 +19,17 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_tree_mat_h)
 #define octave_tree_mat_h 1
 
-#include <iostream>
+#include <iosfwd>
 
 class octave_value;
 class octave_value_list;
 class tree_argument_list;
 
 class tree_walker;
 
 #include "base-list.h"
diff --git a/src/pt.h b/src/pt.h
--- a/src/pt.h
+++ b/src/pt.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_tree_h)
 #define octave_tree_h 1
 
 #include <string>
 
-#include <iostream>
+#include <iosfwd>
 
 class octave_function;
 class tree_walker;
 
 // Base class for the parse tree.
 
 class
 tree
diff --git a/src/utils.h b/src/utils.h
--- a/src/utils.h
+++ b/src/utils.h
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_utils_h)
 #define octave_utils_h 1
 
 #include <cstdarg>
 
-#include <iostream>
+#include <iosfwd>
 #include <string>
 
 #include "dMatrix.h"
 #include "lo-utils.h"
 
 class octave_value;
 class octave_value_list;
 class string_vector;
diff --git a/src/zfstream.cc b/src/zfstream.cc
--- a/src/zfstream.cc
+++ b/src/zfstream.cc
@@ -30,16 +30,18 @@ along with Octave; see the file COPYING.
    original version by Kevin Ruland <kevin@rodin.wustl.edu>
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+#include <iostream>
+
 #include "zfstream.h"
 
 #ifdef HAVE_ZLIB
 
 #include <cstring>          // for strcpy, strcat, strlen (mode strings)
 #include <cstdio>           // for BUFSIZ
 
 // Internal buffer sizes (default and "unbuffered" versions)
diff --git a/src/zfstream.h b/src/zfstream.h
--- a/src/zfstream.h
+++ b/src/zfstream.h
@@ -30,18 +30,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifndef ZFSTREAM_H
 #define ZFSTREAM_H
 
 #ifdef HAVE_ZLIB
 
-#include <istream>  // not iostream, since we don't need cin/cout
-#include <ostream>
+#include <iosfwd>
+
 #include "zlib.h"
 
 /*****************************************************************************/
 
 /**
  *  @brief  Gzipped file stream buffer class.
  *
  *  This class implements basic_filebuf for gzipped files. It doesn't yet support
